### PR TITLE
Refactor: use jquery.cookie -> js-cookie, node-uuid -> uuid

### PR DESCRIPTION
**Pull request by HipsterBrown**
I rebased @aburgel's js-cookie branch against the latest `master` branch and opened a PR off my fork until I get push access to the main repo. 

Some test refactoring provided by me.

### CODE REVIEW

**Comment by HipsterBrown** [comment]
/domain @Betterment/js-platform @aburgel 
/no-platform

nanda

**Comment by chukkwagon** [comment]
<<domain lgtm

File: src/session.test.js [hunk 1]
```diff
@@ -211,14 +216,14 @@ describe('Session', () => {
 
     describe('#signUp()', () => {
       it('updates the visitor id in the cookie', done => {
-        $.cookie = jest.fn();
+        Cookies.set.mockClear();
```
Comment (samandmoore) [inline src/session.test.js]: Isn't there a thing where we can define an after all that will reset all the mocks in jest? And we can do that across the whole suite? I find resetting mocks in individual specs to be an anti pattern.

Comment (HipsterBrown) [reply src/session.test.js]: We have the `jest.config.js` set up to clear mocks after every test, this call here clears it after the session `initialize` method call. I've moved it into the `beforeEach` setup hook.

**Comment by samandmoore** [comment]
lgtm!

**Pull request merged**